src/gui/qgsmaptoolcapture.cpp

```diff
@@ -84,6 +85,8 @@ void QgsMapToolCapture::activate()
     mTempRubberBand->show();
 
   QgsMapToolAdvancedDigitizing::activate();
+
+  mDefaultZValue = QSettings().value( QStringLiteral( "/qgis/digitizing/default_z_value" ), Qgis::DEFAULT_Z_COORDINATE ).toDouble();
```

Comment:
Seems odd to me to store this value in a member variable. Why not just modify getDefaultZValue() to retrieve the settings value? This would allow the default to be changed and immediately reflected.